packages/remix-dev/codemod/utils/git.ts

```diff
@@ -0,0 +1,29 @@
+import execa from "execa";
+
+type GitStatus = "not a git repository" | "clean" | "dirty";
+
+const TEN_MEBIBYTE = 1024 * 1024 * 10;
+export const status = async (
+  dir: string = process.cwd()
+): Promise<GitStatus> => {
+  // Simplified version of `sync` method of `is-git-clean` package
+  try {
+    let gitStatus = await execa("git", ["status", "--porcelain"], {
+      cwd: dir,
+      encoding: "utf8",
+      maxBuffer: TEN_MEBIBYTE,
+    });
```

Comment:
Good point. I do think that `execa` has a nicer API (autotrims output, provides more information about errors) but the main reason was due to it purported improvements for Windows support.

We were considering bundling all the CLI dependencies anyway so that running `npx @remix-run/dev` or `npx create-remix` wouldn't have to subsequently install a bunch of things. So I'll discuss this with the team to see if we think the benefits of `execa` are worth the added dependency.